pkg/osquery/runtime/runner.go

```diff
@@ -218,14 +273,12 @@ func (r *Runner) Interrupt(_ error) {
 func (r *Runner) Shutdown() error {
 	ctx, span := traces.StartSpan(context.TODO())
 	defer span.End()
-
-	if r.interrupted.Load() {
-		// Already shut down, nothing else to do
-		return nil
+	// ensure one shutdown is sent for each instance to read
+	r.instanceLock.Lock()
+	for range r.instances {
+		r.shutdown <- struct{}{}
 	}
```

Comment:
what about giving each instance it's own shutdown channel to avoid the buffered channel?

```suggestion
	for instance := range r.instances {
	        close(instance.shutdown)
	}
```

Reply:
oh yeah i'm gonna give this a try, ty! I'm actually seeing some interesting behavior while trying to write a test for Interrupt during restart per Becca's comment [here](https://github.com/kolide/launcher/pull/2007#discussion_r1932430973) and I wonder if this wouldn't clear things up a little. the behavior I'm seeing in the test is that the interrupt can end up ignored if timed correctly and I think It's related to the buffered channel use so far